game/src/js/objects/ui/Button.js

```diff
@@ -1,21 +1,20 @@
 export default class Button extends Phaser.GameObjects.Text {
-  constructor(scene, x, y, text, options = {}) {
-    super(scene, x, y, text, {
-      fontSize: 30,
-      fontFamily: "'Fjalla One', sans-serif"
-    });
-
-    this.props = {
+  constructor(scene, x, y, text, options = {
+    //use array destructuring to pass default parameters to the child class.
       width: 400,
       disabled: false,
       fontColorNormal: '#d3d3d3',
       fontColorOver: '#ffffff',
       fontColorDisabled: '#333333',
       backgroundColorNormal: '#ffffff00',
       backgroundColorOver: '#ffffff55',
-      backgroundColorDisabled: '#00000000',
-      ...options
-    };
+      backgroundColorDisabled: '#00000000'
+
+    }){
+    super(scene, x, y, text, {
+      fontSize: 30,
+      fontFamily: "'Fjalla One', sans-serif"
+    });
 
     let padding = {left: 5, right: this.props.width - this.width, top: 0, bottom: 0};
```

Comment:
This line will throw an error now that `this.props` is no longer defined.

Reply:
`this.props` is also used below quite a bit. In the constructor closure they could be replaced by `options`, but not in `buttonDown` or `setDisabled`.